File: engine/core/private/src/localization/text_transformer.cpp
/**
 * @file text_transformer.cpp
 *
 * @copyright Copyright (c) 2026 Retro & Chill. All rights reserved.
 * Licensed under the MIT License. See LICENSE file in the project root for full license information.
 */
module retro.core.localization.text_transformer;

import retro.core.localization.localization_manager;

namespace retro
{
    std::u16string TextTransformer::to_lower(std::u16string_view text)
    {
        return LocalizationManager::get().current_locale()->to_lower(text);
    }

    std::u16string TextTransformer::to_upper(std::u16string_view text)
    {
        return LocalizationManager::get().current_locale()->to_upper(text);
    }
} // namespace retro

File: engine/scripting/dotnet/src/RetroEngine.Portable/Localization/TextTransformer.cs
// // @file TextTransformer.cs
// //
// // @copyright Copyright (c) 2026 Retro & Chill. All rights reserved.
// // Licensed under the MIT License. See LICENSE file in the project root for full license information.

using RetroEngine.Portable.Localization.Cultures;

namespace RetroEngine.Portable.Localization;

internal static class TextTransformer
{
    public static string ToUpper(string text)
    {
        return text.ToUpper(CultureManager.Instance.CurrentLocale.CultureInfo);
    }

    public static string ToLower(string text)
    {
        return text.ToLower(CultureManager.Instance.CurrentLocale.CultureInfo);
    }
}
